src/logic/calculate.js

```diff
@@ -0,0 +1,33 @@
+import operate from './operate';
+
+const calculate = (object, buttonName) => {
+  let { total, next, operation } = object;
+
```

Comment:
we are missing the case where user press `0-9`, that's; when `buttonName` is a digit.

You could create an array of 0-9 digits and use `include` with the `if else` statement